parallaxinc/solo
# Solo 111 Fix string var length block

Addresses #111 

**Summary:**
 - Adds fixes from google (added as overrides until we update our core)
 - Some code and variable name cleanup/improvements

**This affects:**
 - wx_scan
 - string_var_length
 - scan_string
 - serial_receive_multiple
 - constant_define

**To Test:**
 - create a new project 
 - make 4 new variables with different names
 - create 2 or 3 named constants with the define constant block
 - drag each of the blocks out and edit them.  Add fields using their mutators.
 - rearrange the blocks in the mutator and make sure the values follow.
 - save and re-open, verify persistence
 - check code view/compile.  Some blocks require inits.
   - FWIW, I'm not worried about code generation - generators were not edited at all, and field names were not changed.
 - Additionally, spot check other blocks with mutators.  The fix from google changes how they render under the hood, so worth a quick check.

**Still investigating:**
Older projects with these blocks being opened back up.  Are the blocks reproduced faithfully?  The XML in those older files looks a bit fishy and I'm suspecting an off-by-1 error.  This would cause them to render as their defaults and not show what the user set them as.  If we see this I can implement a fix, but I'd rather not if we don't have to.
